### PR TITLE
Added rate limiting table and displaying for endpoints

### PR DESCRIPTION
**Pull request by SimonSchick**
Caveats: This pulls the data from `default.js` which shouldn't make a different right now but might later, sadly `config` does not allow being instantiated with different working directories.

Pro: Automatic rate limit table building, YAY!

@connor4312 If you have a minute it would be totally cool if you could add a little `database` or `cache` icon to the endpoints (like the `lock` icon for secured endpoints), I didn't really understand how your glyph icon font thingy works 😛 

Requires: https://github.com/WatchBeam/backend/pull/859 to be merged to master to work correctly.

### CODE REVIEW

File: build/misc.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ const http = require('http');
...
 const _ = require('lodash');
+const orderObject = require('./util').orderObject;
+
+/**
+ * Reads a json file
+ * @param  {String} filePath
+ * @return {Object}
+ */
+function readJSONFile (filePath) {
+    return JSON.parse(fs.readFileSync(filePath));
```
Comment (JamyDev) [inline build/misc.js]: `require(filePath)` ? :P

File: src/rest/resource.pug [hunk 2]
```diff
@@ -54,6 +54,12 @@ mixin method(method, resource)
                                         pre.oauth-scope= scope
                                 if securityScheme.description
                                     | !{marked(securityScheme.description)}
+                    if method.is
+                        - rateLimitInfo = restUtil.getTraitInfo(method, 'rateLimited')
+                        - console.log(rateLimitInfo)
```
Comment (JamyDev) [inline src/rest/resource.pug]: Do we really want to log this all the time?

**Comment by JamyDev** [comment]
lgtm

File: build/misc.js (hunk 1 above)
Comment (connor4312) [inline build/misc.js]: cache-busting

**Comment by connor4312** [comment]
Re to the conifg: what I would do is either:
 - submit a PR to node-config to add subdirectory / instanced loading or
 - create a tiny standalone module that requires the config module and outputs the config to stdout before exiting. Then you can subprocess that module and set the working directory appropriately. This is a slightly ugly solution but isn't too hacky and should be stable enough.

**Comment by SimonSchick** [comment]
I'll look into creating a PR for config since it seems like a pretty straight-forward process.

**Comment by SimonSchick** [comment]
Can we get this merged for now? We want to roll out rate limiting so it would be pretty good to have this in.

**Comment by ProbablePrime** [comment]
Sure go ahead and merge it, i'll get you an icon. I also need to add an icon for deprecation. feature/FANCY ICONS incoming soon :)

**Pull request merged**